### PR TITLE
Scatter correlations

### PR DESCRIPTION
**Pull request by d-callan**
in support of bpnet 2.0 scatter plots

### CODE REVIEW

File: tests/testthat/test-scattergl.R [hunk 1]
```diff
@@ -1384,12 +1384,12 @@ test_that("scattergl.dt() returns correct information about missing data", {
...
-  expect_equal(names(jsonList$scatterplot$data),c('facetVariableDetails','seriesX','seriesY','seriesGradientColorscale'))
+  expect_equal(names(jsonList$scatterplot$data),c('facetVariableDetails','seriesX','seriesY','seriesGradientColorscale','correlationCoef','pValue'))
   expect_equal(names(jsonList$scatterplot$data$facetVariableDetails[[1]]),c('variableId','entityId','value'))
   expect_equal(length(jsonList$scatterplot$data$facetVariableDetails), 10)
   expect_equal(jsonList$scatterplot$data$facetVariableDetails[[1]]$variableId, 'cat4')
   expect_true(all(is.na(jsonList$scatterplot$data$seriesGradientColorscale[[2]]))) # should be null in the json file, but fromJSON converts to NA
-  expect_equal(names(jsonList$scatterplot$config),c('variables','completeCasesAllVars','completeCasesAxesVars'))
+  expect_equal(names(jsonList$scatterplot$config),c('variables','completeCasesAllVars','completeCasesAxesVars','correlationMethod'))
   expect_equal(names(jsonList$scatterplot$config$variables$variableSpec),c('variableId','entityId'))
   expect_equal(jsonList$scatterplot$config$variables$variableSpec$variableId[jsonList$scatterplot$config$variables$plotReference == 'overlay'], 'contC')
```
Comment (asizemore) [inline tests/testthat/test-scattergl.R:1394]: We should also test to make sure we're getting back at least some results and that they're of the right type.

File: R/class-plotdata-scatter.R [hunk 2]
```diff
@@ -103,6 +117,14 @@ newScatterPD <- function(.dt = data.table::data.table(),
   } else {
     .pd <- series
   }
+
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/class-plotdata-scatter.R:120]: yea i realized while doing this that i dont love how we handle these cases currently.. looking at mosaic, you can say which stats you want but cant turn stats off there and just get counts. then looking at box, you can toggle stats on and off but cant say which stats you want, though i recognize thats maybe mostly due to there not really being an option yet. but even then id still like it being explicit i think.. anywho, i followed mosaics example... but maybe should have tried to do better.

adding a second boolean arg for computeStats could potentially be confusing.. and i think if we try to use NULL as an option for the existing correlationMethod arg then matchArg will just default to the first value. so i guess were left w adding a 'none' value to correlationMethod, and handling that as a special case.. which isnt ideal but is maybe better than the other options ive thought of.

File: R/class-plotdata-scatter.R [hunk 3]
```diff
@@ -30,18 +31,31 @@ newScatterPD <- function(.dt = data.table::data.table(),
   group <- veupathUtils::findColNamesFromPlotRef(variables, 'overlay')
   panel <- findPanelColName(veupathUtils::findVariableSpecFromPlotRef(variables, 'facet1'), 
                             veupathUtils::findVariableSpecFromPlotRef(variables, 'facet2'))
+  
+  dtForCorr <- data.table::copy(.pd)
+  attributes(dtForCorr)$class <- c('data.table', 'data.frame')
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/class-plotdata-scatter.R:36]: cause im dumb

File: tests/testthat/test-scattergl.R [hunk 4]
```diff
@@ -772,7 +772,7 @@ test_that("scattergl.dt() returns an appropriately sized data.table", {
 
   dt <- scattergl.dt(df, variables, 'raw')
   expect_equal(nrow(dt), 9)
-  expect_equal(names(dt), c('entity.cat3', 'entity.factor3','seriesX','seriesY'))
+  expect_equal(names(dt), c('entity.factor3', 'entity.cat3','seriesX','seriesY', 'correlationCoef', 'pValue'))
```
[the remark this answers is not in the document]
Comment (d-callan) [reply tests/testthat/test-scattergl.R:775]: idk tbh

File: R/class-plotdata-scatter.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-callan) [reply R/class-plotdata-scatter.R:120]: i guess ill have it default to none as well..

**Pull request merged**